### DISCUSSION TITLE
Where do I find ltverify.class?

### DISCUSSION THREAD

**Discussion by JeffWeyant** [appcelerator/titanium_mobile#13524] [Q&A]
I'm trying to use Xcode to attach an Apple Distribution Profile to an app previously created in Appcelerator, and get a message that it can't find org/appcelerator/titanium/tiverify.class.

The app was created on an old MacMini, using Appcelerator Studio.  I can't create a Proifle with the version of XCode installed there, and can't upgrade XCode, because that MacMini is too old. I copied the workspace from the old MacMini to the new one, with the latest version of XCode, but when I try to build it, I get the error: "library not found for 'ltiverify".  I assume this is because I need to copy over more than just my project workspace, but can't seem to find it.

**Comment by m1ga** [comment]
Hi, which SDK version are you using? 
Normally you shouldn't use XCode to build the app. There is this PR https://github.com/tidev/titanium_mobile/pull/13032 that will fix some Xcode stuff so you can build apps again.
Do you have Titanium installed on the new machine? If so run `ti build -p ios --target ?` and select your profile using the menu. Everything you'll need is in the workspace/application-name folder. 

I'm not sure if tiverify is still used if you trying to build it with SDK 11. Isn't that a module that verified modules before?

**Reply by JeffWeyant** [reply]
My preference wasn't to use XCode, but I think I'm kind of forced into it, unless there is an alternative I'm not aware of.  The program was written in a version of Appcelerator that was installed on an old MacMini.  I can't use that one for two reasons. 1) I can no longer login, I get the error "getaddrinfo ENOTFOUND", which I'm guessing is because with Appclerator Studio reaching end-of-life, all credentials have been removed, and 2) when I tried to create the profile on that MacMini, I was told I can't because the current version of XCode doesn't support it.  (Sorry, but without being able to log in, I'm not sure which SDK, but I don't see that as an issue at this point since I can't use it).  I tried to update the XCode version on the old MacMini, but couldn't because that MacMini is no longer supported.

So...  I have a new MacMini, and installed the latest version of XCode.  I am now able to create the new Distribution Profile.  I had tried a couple years ago to install Appcelerator Studio on that new MacMini, but found out it couldn't because it has the new M1 chip, which Appcelerator Studio didn't like.  That's the reason for hoping to use XCode to build the app with the new profile.

I copied all the project files for the project from the old MacMini to the new MacMini.  When I try to build, I get the error "Library not found -ltverify".  Not really sure what that is, or if I need it.

I tried the ti build -p ios --target ? you suggested, but it says no matches found on the question mark.  Was I supposed to replace that with something?

**Reply by m1ga** [reply]
Just means you are using zsh as a terminal. So you have to use # instead of ?
You can just use VSCode to open your project or look into the tiapp.xml with a text editor to check the version. The whole app project is a folder that you can open/browse without Appcelerator Studio. And VSCode has a Titanium extension that allows you to build the project with a GUI so you don't need the CLI. 
But the CLI is easier since you don't need to setup VSCode + extension. https://fromzerotoapp.com/how-to-install-appcelerator-titanium/ if you'll need to install everything on your new machine (basically `sudo npm i -g titanium alloy` and you are good to go).

**Reply by JeffWeyant** [reply]
Tried the sudo...  command, and got Permission Denied errors (which I find surprising, as I'm the only user/administrator for this MacMini).  Is there a list of CLI commands I could look at, to see how to attach the new Distribution Profile, and re-build the app with it?  I don't need to make any code changes, just take what was in my old Appcelerator project file, and build it with the new Distribution Profile.

**Reply by m1ga** [reply]
That I don't know, would be more a Xcode questions outside of Titanium, so you could search for: Xcode project profile change on google. Is it an internal app? Otherwise you'll need to update the version number.
If the npm command is not working you should perhaps check if node/npm is installed correctly. Sounds like something is wrong on your machine :) But again: outside of the Titanium scope, you'll find better tutorials by searching for "m1 install node". 

For Titanium you only need the `ti` command from the first post if everything is installed to rebuild the app and be able to select the profile. For Xcode I don't know it since I never used it standalone and I don't know how to resign an existing project - which I know is possible but I never had to do it either.

**Reply by JeffWeyant** [reply]
Looks like the M1 MacMini is going to be a hassle.  Figured I would copy the Distribution Profile I was able to create there back to the old MacMini, and try to build there, since Appcelerator Studio already exists there.  I tried the ti build -p ios --target ? and got:

[ERROR] Failed to load command "build"
[ERROR] TypeError: error.stack.split is not a function.

I have Appcelerator Studio installed, but it's the old one from Axway.  I was hoping the ti commands would still work.  Is this a case of me needing to use the newest version of Titanium, not the one that was previously Axway's Appcelerator?

Thanks again for all the help.

**Reply by m1ga** [reply]
it would help to know which Ti version you are using. Open the tiapp.xml file with a text editor and check the `<sdk-version>` node.

There is only an old Studio. There isn't any newer non Axway version. 

`ti info` to get you version infos. And `ti help` for the commands. I think some really old versions used `ti run` to build?

I still think it is way easier to install it on the new machine (https://fromzerotoapp.com/how-to-install-appcelerator-titanium/). If you going to upload the app to the app store you have to use Xcode 13

**Reply by JeffWeyant** [reply]
I've tried installing it on the new machine, but I still get the EACCES error:

jeffreyweyant@Jeffreys-Mini ~ % npm i -g titanium alloy
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm WARN deprecated [har-validator@5.1.5](mailto:har-validator@5.1.5): this library is no longer supported
npm WARN deprecated [sprintf@0.1.5](mailto:sprintf@0.1.5): The sprintf package is deprecated in favor of sprintf-js.
npm WARN deprecated [nomnom@1.8.1](mailto:nomnom@1.8.1): Package no longer supported. Contact [support@npmjs.com](mailto:support@npmjs.com) for more info.
npm WARN deprecated [uuid@3.4.0](mailto:uuid@3.4.0): Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated [request@2.88.2](mailto:request@2.88.2): request has been deprecated, see https://github.com/request/request/issues/3142
npm ERR! code EACCES
npm ERR! syscall symlink
npm ERR! path ../lib/node_modules/alloy/bin/alloy
npm ERR! dest /usr/local/bin/alloy
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, symlink '../lib/node_modules/alloy/bin/alloy' -> '/usr/local/bin/alloy'
npm ERR!  [Error: EACCES: permission denied, symlink '../lib/node_modules/alloy/bin/alloy' -> '/usr/local/bin/alloy'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'symlink',
npm ERR!   path: '../lib/node_modules/alloy/bin/alloy',
npm ERR!   dest: '/usr/local/bin/alloy'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jeffreyweyant/.npm/_logs/2022-08-11T19_57_32_635Z-debug-0.log
jeffreyweyant@Jeffreys-Mini ~ %

Looks like a permissions error, but I'm the only user of this MacMini.  I'm sure it's probably something simple, but I'm Mac challenged, (we're actually a Windows shop, and only use the Mac to compile the iOS packages).

**Reply by m1ga** [reply]
you didn't use `sudo` to run the command. Otherwise https://stackoverflow.com/a/61646294/5193915

**Reply by JeffWeyant** [reply]
Ahhh, thanks.  Didn't realize the command on the fromzerotoapp was missing that.  Do I preface all the commands they list with sudo?  Or just that one?

**Reply by m1ga** [reply]
I'll change that, it's my site :smile: If you want to install npm packages that you want to access for all users you run it with `sudo npm i -g` the rest of the `ti` commands will be local commands. So it's just the install part from npm

**Reply by JeffWeyant** [reply]
If I'm trying to build from the command line, can I just remove the references to the Appcelerator GUID, rather than trying to generate a different one?

**Reply by m1ga** [reply]
never tried it without. I just use a random one from the url in the tutorial or `uuidgen` in a terminal

**Comment by JeffWeyant** [comment]
Thanks!  I think I've almost got it.  Got everything installed.  Changed the target SDK in tiapp.xml to the installed 11.0.0.  Incremented the app's version number.  

It started compiling, and finally got the error:
[ERROR] titanium_prep failed to run (10)
[ERROR] This application cannot be built with the Titanium open source SDK because it is an Appcelerator Platform registered application. Please use the Appcelerator Platform CLI tools or Appcelerator Studio to build this application.

Is there something I can edit in the tiapp.xlm to get around this?

**Reply by m1ga** [reply]
sure, I have a tutorial for almost everything :smile:  https://fromzerotoapp.com/transfer-your-app-from-appc-cli-to-ti-cli/
change the GUID

**Reply by JeffWeyant** [reply]
Nice site!  I seem to have the compilation working now, and now just have what looks like a certificate issue when it's done:

[TRACE] error: No profile for team 'C595F4F3RW' matching 'Fluor SAs Distribution Profile 20230207' found: Xcode couldn't find any provisioning profiles matching 'C595F4F3RW/Fluor SAs Distribution Profile 20230207'. Install the profile (by dragging and dropping it onto Xcode's dock item) or select a different one in the Signing & Capabilities tab of the target editor. (in target 'Fluor SAs' from project 'Fluor SAs')
[ERROR] 2022-08-13 13:40:27.110 xcodebuild[22102:1603205] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
[ERROR] 2022-08-13 13:40:27.110 xcodebuild[22102:1603205] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
[ERROR] ** ARCHIVE FAILED **

Not sure what that [TRACE] error is telling me.  I've checked XCode, and the team, profile, and certificate are all correct.

**Reply by m1ga** [reply]
hmmm...not that familiar with iOS errors but I would just create a new certificate and provisioning profile and import that on your machine. Just to make sure all are fine. There are some google hits when you search for those errors e.g. https://fluttercorner.com/requested-but-did-not-find-extension-point-with-identifier/ or this https://developer.apple.com/forums/thread/703233?answerId=711300022#711300022 (and following posts)

**Comment by JeffWeyant** [comment]
Yeah, I tried suggestions on those sites, and a couple others, but to no avail.  One thing I did notice...  On the original MacMini, the Bundle ID was built as a wildcard:  com.fluor.*  It looks like after installing XCode on the new MacMini, and copying over all the project files, the Bundle ID showing up in XCode is com.fluor.fluorsas  (fluorsas is the name of the project).  Do I need to go through XCode, and change references of com.fluor.fluorsas to com.fluor.*?  In order to match what's on the certificate?

**Reply by m1ga** [reply]
You can get them all from the Apple Developer page. The certificates expire after some time. So if you just copy them over from an old machine they are probably out of date. Wildcards are mostly for testing. If you want to build for the store you have to use the specific file.

**Comment by JeffWeyant** [comment]
It's for ad-hoc deployment through our MDM, not the Apple Play Store.  The certificate and profile are both current, (the profile is brand new, that's what started me on this journey, heh).  I've downloaded everything from the Apple Developer page, just didn't know if I needed to change the specific Bundle ID in XCode to match the wildcard that was used on the original release, (since that is the App ID used in creating the Provisioning Profile)

**Reply by m1ga** [reply]
I see, sorry got a bit lost in all the posts :smile: I only use wildcards for dev stuff and matching for deploying on device. But I'm the wrong person for iOS stuff. I'm always glad it works on iOS myself and I don't have to deal with special cases. And I've never done any manuel xcode stuff so don't know that, sorry. Maybe try asking in Slack https://tislack.org since there are more users watching then here :-)

**Reply by JeffWeyant** [reply]
Thanks, I'll check with them.  Your site, and all your help, was instrumental in getting me to this point though.  I really appreciate all the help!!
